.github/workflows/topology_tests.yml

```diff
@@ -0,0 +1,37 @@
+on:
+  pull_request:
+    paths:
+      - 'chart/**'
+      - 'test/**'
+      - '**.go'
+      - '.github/workflows/topology_tests.yml'
+name: Test Topology
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - uses: azure/setup-helm@v3
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }} # only needed if version is 'latest'
+        id: install
+      - name: Install Karina
+        run: make .bin/karina
+      - name: Build and Cache Image
+        run: |
+          # pull image if exists
+          echo ${{ secrets.GITHUB_TOKEN }} | docker login ghcr.io -u $GITHUB_ACTOR --password-stdin
+          docker pull ghcr.io/$GITHUB_REPOSITORY/canary-checker:ci || true
```

Comment:
```suggestion
          docker pull docker.io/flanksource/canary-checker:ci || true
```

Reply:
This is image is meant for cache

Should we be putting it on dockerhub?

Reply:
Yes, we dont have gcp keys to push to ghcr

Reply:
n.m I though it was gcr